usr/share/rear/finalize/Linux-i386/630_run_efibootmgr.sh

```diff
@@ -9,7 +9,7 @@ is_true $EFI_STUB && return 0
 
 # UEFI_BOOTLOADER empty or not a regular file means using BIOS cf. rescue/default/850_save_sysfs_uefi_vars.sh
 # Double quotes are mandatory here because 'test -f' without any (possibly empty) argument results true:
-test -f "$UEFI_BOOTLOADER" || return 0
+test -f "$TARGET_FS_ROOT/$UEFI_BOOTLOADER" || return 0
```

Comment:
This code does the right thing but in a subtle way
because when UEFI_BOOTLOADER is empty (which means BIOS)
```
test -f "$TARGET_FS_ROOT/$UEFI_BOOTLOADER"
```
evaluates to
```
test -f "/mnt/local/"
```
which is not true because `/mnt/local/` is a directory.

Reply:
Good point!
I did not think about that ...

V.

Reply:
ah, first I thought that I did it wrong :-), honestly I did not realized that `-f` will evaluate directory as false :-) ...

V.

Reply:
That's why explanatory comments help so much ;-)

I think they help not only others to understand how it is meant to work.
I think they help (perhaps even more) the one who makes the code
to get things sorted out in his mind.

At least this is what happpens to me when I make code
and then try to explain what I did.
When it gets hard for me to explain my own code I usually got
something wrong or at least I can make my code better
until I can well explain what my code does.
This is the reason behind why I make so much comments.